### ISSUE TITLE
Swift Command Line tool & CocoaPods issue

### ISSUE THREAD

**Issue by andyzaharia** [SVGKit/SVGKit#681]
Hi guys, first of all, thank you to everyone that contributed to this awesome framework! 

Now to my issue .... I'm trying to integrate SVGKit into a Command Line too via CocoaPods / static libraries. CocoaPods seems to handle everything correctly from what I can tell, but when I try to use the framework I get a "No such module ''SVGKit" error. 

Because I'm using Swift, I've added a bridging header where I import the Objective-C code / framework, I tried #import <SVGKit/SVGKDefine.h> and #import <SVGKit/SVGKit.h> with no luck.

I also added the framework path into the Framework Search Paths and it still doesnt want to find the "module". 

Can this actually be a problem with the fact that I'm actually trying to build a command line tool and this wont work at all with SVGKit?

Anyone has any ideas? Thinking about it... is this even possible to achieve? 

Thanks a lot!

**Comment by andyzaharia** [comment]
Closing this one as it seems to be not possible to include the framework inside a command line tool.

**Comment by adamgit** [comment]
I'm surprised it wouldn't work - is there something special in command-line
apps that blocks libraries?

Did you try it without CocoaPods? CocoaPods has a long history of
breaking/blocking things exactly like this one (which is why we don't
recommend it).

**Comment by andyzaharia** [comment]
Not entirely sure why they are different, maybe the fact that it cannot include bundles and can link only static libraries... I've read that even the carthage executable is generated/extracted from an app bundle to avoid the command line limitations, I really didnt dig too long into this one as it seems to require quite a few workarounds.

Technically, it should not be a problem to get it working as long as we can compile SVGKit as a static library and link it. 

With Carthage I was getting "dyld: Library not loaded", I've played a bit with the build settings but couldnt get it fixed in a timely manner...

There is more info here https://github.com/Carthage/Carthage/issues/636 

I will try to get it working when I get some more free time to play with it.

**Comment by adamgit** [comment]
Off the top of my head, SVGKit originally was static library only - Apple
didn't allow iOS apps to have anything else - if so, we had this working
for a long time (although I remember it was difficult to maintain -
contributors wouldn't often accidentally overwrite the linker settings
because xcode gave them no warning, or because poorly-written tutorials on
the web told them to without explanation).

Nearly always it was a problem with the various header-paths, object-paths,
and the presence/abscene of the flags "-all_load" and friends (can't
remember them all, but there was a triumvirate around all-load that often
went hand in hand).

**Comment by andyzaharia** [comment]
Yea, the static library compilation works just fine both with CocoaPods and Carthage, the problem is something to do at the linking stage. As you said it must be something with the flags / search paths. 

Thanks a lot Adam for your input! I will try to get back when I get more time to investigate this and play with the build settings. 

And thank you to everyone who contributed to this nice framework!

**Comment by adamgit** [comment]
I'm sorry I couldn't be more specific :( - it's been more than 5 years
since I did anything with the static library settings I think.

I remember spending a lot of time carefully reading the xcode
developer-docs on Apple.com to find out EXACTLY how Apple used each of the
different various search-paths, and what was legal/illegal in them (there
are several subtleties there, and xcode simply sielntly fails if it doesn't
like them).

I also remember LOTS of bugs in Xcode in this area - some hilariously bad
ones e.g. where Apple did a search/replace on specific chars in your search
path and just literally wiped search paths for no reason ... Apple staff
shiping stuff in production without testing :) ... so you had to put paths
in once, click away from the field, click back (for Xcode to corrupt it),
then edit it a second time to make the "real" value stick. Oh what fun
times those were!

TL;DR: there may still be some amazingly dumb bugs in Xcode there (although
all the ones I remember ... eventually got fixed). Comparing what you think
you typed in a field to what's actually there, or to what Xcode puts on the
commandline when it does a Build Log often quickly finds whatever has gone
wrong.
